media/js/base/mozilla-client.js

```diff
@@ -10,11 +10,84 @@ if (typeof Mozilla === 'undefined') {
 (function() {
     'use strict';
 
+    /**
+     * Provide information on the user's browsing environment, including the platform and browser details.
+     *
+     * @namespace
+     * @see {@link https://developer.mozilla.org/en-US/docs/Gecko_user_agent_string_reference}
+     */
     var Client = {};
 
     /**
-     * Get the user's Firefox version number. '0' will be returned on non-Firefox browsers.
+     * Detect whether the user's browser is Firefox on any platform. This includes WebKit-based Firefox for iOS.
+     *
+     * @private
+     * @param  {String} ua - browser's user agent string, navigator.userAgent is used if not specified
+     * @return {Boolean} result
+     */
+    Client._isFirefox = function (ua) {
+        ua = ua || navigator.userAgent;
+
+        return /\s(Firefox|FxiOS)/.test(ua) && !Client._isLikeFirefox(ua);
+    };
+
+    /**
+     * Detect whether the user's browser is Firefox for Windows, OS X or Linux.
+     *
+     * @private
+     * @param  {String} ua - browser's user agent string, navigator.userAgent is used if not specified
+     * @return {Boolean} result
+     */
+    Client._isFirefoxDesktop = function (ua) {
+        ua = ua || navigator.userAgent;
+
+        return /\sFirefox/.test(ua) && !/Mobile|Tablet|Fennec/.test(ua) && !Client._isLikeFirefox(ua);
+    };
+
+    /**
+     * Detect whether the user's browser is Firefox for Android.
      *
+     * @private
+     * @param  {String} ua - browser's user agent string, navigator.userAgent is used if not specified
+     * @return {Boolean} result
+     */
+    Client._isFirefoxAndroid = function (ua) {
+        ua = ua || navigator.userAgent;
+
+        return /\sFirefox/.test(ua) && /Mobile|Tablet|Fennec/.test(ua) && !Client._isLikeFirefox(ua);
```

Comment:
From the tests it looks like this returns true for Firefox OS? Would this be more robust if we we're checking for `android` specifically?